OpenDreamRuntime/Objects/Types/DreamObjectException.cs

```diff
@@ -1,26 +1,47 @@
 namespace OpenDreamRuntime.Objects.Types;
 
 public sealed class DreamObjectException(DreamObjectDefinition objectDefinition) : DreamObject(objectDefinition) {
-    public string Name = string.Empty;
-    public string Description = string.Empty;
-    public string File = string.Empty;
-    public int Line = 0;
+    public DreamValue Name = DreamValue.Null;
+    public DreamValue Desc = DreamValue.Null;
+    public DreamValue File = DreamValue.Null;
+    public DreamValue Line = DreamValue.False;
 
     //TODO: Match the format of BYOND exceptions since SS13 does splittext and other things to extract data from exceptions
 
+    protected override void SetVar(string varName, DreamValue value) {
+        switch (varName) {
+            case "name":
+                Name = value;
+                return;
+            case "desc":
+                Desc = value;
+                return;
+            case "file":
+                File = value;
+                return;
+            case "line":
+                Line = value;
+                return;
```

Comment:
This misses the part where DreamObjects that aren't lists will instead use that object's `GetDisplayName()`

Reply:
Ignore me, misunderstood the output of `json_encode()`